src/frontend/cfg_parser.ml

```diff
@@ -505,10 +504,24 @@ let rec edge_list_of_stmt method_id loc_map entry exit ret exc ?(brk = None) stm
             expr ~curr_loc:entry ~exc e
       in
       (loc_map, (entry, exit, Stmt.Assume expr) :: intermediate_stmts)
-  | `Blk (_, stmts, _) -> edge_list_of_stmt_list method_id loc_map ~entry ~exit ~ret ~exc ~brk stmts
-  | `Brk_stmt (_, None, _) -> (loc_map, [ (entry, Option.value_exn brk, Stmt.Skip) ])
-  | `Brk_stmt _ -> unimplemented "`Brk_stmt with label" (loc_map, [])
-  | `Cont_stmt _ -> unimplemented "`Cont_stmt" (loc_map, [])
+  | `Blk (_, stmts, _) ->
+      edge_list_of_stmt_list method_id loc_map ~entry ~exit ~ret ~exc ~brk ~cont stmts
+  | `Brk_stmt (_, None, _) -> (
+      match fst brk with
+      | Some brk_target -> (loc_map, [ (entry, brk_target, Stmt.Skip) ])
+      | None -> unimplemented "`Brk_stmt inside try catch with finally" (loc_map, []) )
+  | `Brk_stmt (_, Some (_, label), _) -> (
+      match List.find (snd brk) ~f:(fst >> snd >> ( = ) label) with
```

Comment:
This seems more complicated than needed -- why not just store a string in the `~brk`/`~cont` parameter instead of a `CST.void_type`?

Reply:
`~f` here would then become `fst >> String.equal label`